src/rules/noMagicNumbersRule.ts

```diff
@@ -120,16 +120,14 @@ function parseOptions(options: Array<OptionsType | number>): ParsedOptionsType {
     for (const option of options) {
         if (typeof option === "number") {
             parsedOptions[ALLOWED_NUMBERS_OPTION].push(option);
-        }
-        if (option.hasOwnProperty(ALLOWED_NUMBERS_OPTION)) {
+        } else if (option.hasOwnProperty(ALLOWED_NUMBERS_OPTION)) {
             const numberOptions = (option as AllowedNumbersType)[ALLOWED_NUMBERS_OPTION];
             if (Array.isArray(numberOptions) && numberOptions.length > 0) {
                 numberOptions.forEach((num: number) =>
                     parsedOptions[ALLOWED_NUMBERS_OPTION].push(num),
                 );
             }
-        }
-        if (option.hasOwnProperty(IGNORE_JSX_OPTION)) {
+        } else if (option.hasOwnProperty(IGNORE_JSX_OPTION)) {
```

Comment:
Or is a switch statement of preference?

Reply:
How about a `continue` on the inside of the `if (typeof option === "number")` case?

It's possible an option object would have both `ALLOWED_NUMBERS_OPTION` and `IGNORE_JSX_OPTION`.